### PR TITLE
Weather app

### PR DESCRIPTION
**Pull request by KaralyosBela**

### CODE REVIEW

File: react-app/src/pages/WeeklyWeatherPage.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const WeeklyWeatherPage: React.FC = () => {
           `https://api.openweathermap.org/data/2.5/onecall?lat=33.44&lon=-94.04&appid=${process.env.REACT_APP_API_KEY}&units=metric`
         );
 
-        //TODO: itt mit kéne
+        //TODO: az apinak kellene interface, és az lenne a type az itemnek
         const asd = response.data.daily.map((item: any) => {
```
Comment (martonszel) [inline react-app/src/pages/WeeklyWeatherPage.tsx:19]: item : IWeeklyWeather ?

Comment (martonszel) [inline react-app/src/pages/WeeklyWeatherPage.tsx:19]: asd not a descriptive name

File: react-app/src/pages/WeeklyWeatherPage.tsx [hunk 2]
```diff
@@ -32,6 +32,10 @@ const WeeklyWeatherPage: React.FC = () => {
     fetchWeatherApi();
```
Comment (martonszel) [inline react-app/src/pages/WeeklyWeatherPage.tsx:32]: no console logs

File: react-app/src/interfaces/Weather.interface.ts [hunk 3]
```diff
@@ -8,4 +8,11 @@ export interface IWeeklyWeather {
   temperature: string,
   description: string,
   icon: string
+}
+
+export interface ICity {
```
Comment (martonszel) [inline react-app/src/interfaces/Weather.interface.ts:13]: what does it I mean in ICity and Iweekly... ?

**Review by martonszel — APPROVED** [review]

File: react-app/src/pages/WeeklyWeatherPage.tsx (hunk 1 above)
Comment (KaralyosBela) [reply react-app/src/pages/WeeklyWeatherPage.tsx:19]: The object stored in the response has so many properties that i did not created an interface to match it.

Comment (szkertesz) [reply react-app/src/pages/WeeklyWeatherPage.tsx:19]: You don't have to cover all of the properties available on the object, you can select only the ones you use. If I'm not wrong.

File: react-app/src/pages/WeatherPage.tsx [hunk 4]
```diff
@@ -1,44 +1,79 @@
...
 import Container from "../components/Container";
 import DailyWeather from "../components/DailyWeather";
-import {Weather} from "../interfaces/Weather.interface";
+import { Weather, ICity } from "../interfaces/Weather.interface";
 import "../styles/Custom.module.css";
 
 const WeatherPage: React.FC = () => {
   const [weather, setWeather] = useState<Weather | null>(null);
+  const [cityData, setCityData] = useState<ICity | null>(null);
+  const [error, setError] = useState<boolean>(false);
```
Comment (szkertesz) [inline react-app/src/pages/WeatherPage.tsx:11]: What is the purpose of this `error` state? I don't see any application of it anywhere -- but I might be blind!

File: react-app/src/pages/WeatherPage.tsx [hunk 5]
```diff
@@ -1,44 +1,79 @@
...
+  // const [cityName, setCityName] = useState<string>("");
+  const cityRef = useRef<HTMLInputElement>(null);
+
+  const searchCity = async () => {
+    //object possibly null
+    const cityName = cityRef.current?.value || "Debrecen";
```
Comment (szkertesz) [inline react-app/src/pages/WeatherPage.tsx:17]: This optional chaining is cool. I should use it too.

**Review by szkertesz — APPROVED** [review]
Nice app!

File: react-app/src/pages/WeatherPage.tsx (hunk 4 above)
Comment (KaralyosBela) [reply react-app/src/pages/WeatherPage.tsx:11]: I did a conditional rendering in the html render section based on the `error` state.

**Pull request merged**